internal/batches/service/remote.go

```diff
@@ -15,6 +15,96 @@ func (svc *Service) areServerSideBatchChangesSupported() error {
 	return nil
 }
 
+const upsertEmptyBatchChangeQuery = `
+mutation UpsertEmptyBatchChange(
+	$name: String!
+	$namespace: ID!
+) {
+	upsertEmptyBatchChange(
+		name: $name,
+		namespace: $namespace
+	) {
+		name
+	}
+}
+`
+
+func (svc *Service) UpsertBatchChange(
+	ctx context.Context,
+	name string,
+	namespaceID string,
+) (string, error) {
+	if err := svc.areServerSideBatchChangesSupported(); err != nil {
+		return "", err
+	}
+
+	var resp struct {
+		UpsertEmptyBatchChange struct {
+			Name string `json:"name"`
+		} `json:"upsertEmptyBatchChange"`
+	}
+
+	if ok, err := svc.client.NewRequest(upsertEmptyBatchChangeQuery, map[string]interface{}{
+		"name":      name,
+		"namespace": namespaceID,
+	}).Do(ctx, &resp); err != nil || !ok {
+		return "", err
+	}
+
+	return resp.UpsertEmptyBatchChange.Name, nil
+}
+
+const createBatchSpecFromRawQuery = `
+mutation CreateBatchSpecFromRaw(
+    $batchSpec: String!,
+    $namespace: ID!,
+    $allowIgnored: Boolean!,
+    $allowUnsupported: Boolean!,
+    $noCache: Boolean!,
+) {
+    createBatchSpecFromRaw(
+        batchSpec: $batchSpec,
+        namespace: $namespace,
+        allowIgnored: $allowIgnored,
+        allowUnsupported: $allowUnsupported,
+        noCache: $noCache,
+    ) {
+        id
+    }
+}
+`
+
+func (svc *Service) CreateBatchSpecFromRaw(
```

Comment:
Should we still keep `UpsertBatchSpecInput` in the service layer here if no one's using it anymore? No strong feelings, just a thought.

Reply:
I have no strong feelings about.I'll remove it.